src/catalog_api.cpp

```diff
@@ -172,8 +174,11 @@ static unique_ptr<HTTPResponse> GetTableMetadata(ClientContext &context, IRCatal
 	url_builder.AddPathComponent("tables");
 	url_builder.AddPathComponent(table);
 
-	auto url = url_builder.GetURL();
-	return catalog.auth_handler->GetRequest(context, url_builder);
+	HTTPHeaders headers(*context.db);
+	if (catalog.attach_options.access_mode == IRCAccessDelegationMode::VENDED_CREDENTIALS) {
+		headers.Insert("X-Iceberg-Access-Delegation", "vended-credentials");
```

Comment:
I still think this really shouldn't be required to be omitted, given the spec:
> Optional signal to the server that the client supports delegated access
via a comma-separated list of access mechanisms.  The server may choose
to supply access via any or none of the requested mechanisms.

But fine, the real world is different from the spec